monitoring/os_linux_test.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2017 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitoring
+
+import (
+	"context"
+	"io"
+	"io/ioutil"
+	"strings"
+
+	"github.com/gravitational/satellite/agent/health"
+	pb "github.com/gravitational/satellite/agent/proto/agentpb"
+	"github.com/gravitational/satellite/lib/test"
+
+	. "gopkg.in/check.v1"
+)
+
+func (*MonitoringSuite) TestValidatesOS(c *C) {
+	// setup
+	var testCases = []struct {
+		releases   []OSRelease
+		getRelease osReleaseGetter
+		probes     health.Probes
+		comment    string
+	}{
+		{
+			releases:   staticOSReleases("ubuntu", "16.04.3", "centos", "7.2"),
+			getRelease: testGetOSRelease(OSRelease{ID: "CentOS", VersionID: "7.2"}),
+			probes:     health.Probes{&pb.Probe{Checker: osCheckerID, Status: pb.Probe_Running}},
+			comment:    "requirements match",
+		},
+		{
+			releases:   staticOSReleases("ubuntu", "16", "centos", "7.2"),
+			getRelease: testGetOSRelease(OSRelease{ID: "ubuntu", VersionID: "16.04.3"}),
```

Comment:
I'm sure this is rooted somewhere else, but I find this sort of interesting. Centos/Redhat are including sub releases (7.2) but ubuntu is only matching the year of the release (16)? For example, at a certain point in time ubuntu 16.04 would be LTS, but you would also have 16.10 version.

Reply:
I think the test intention here is to verify that if the requirement is just the release number (`16` for Ubuntu, regardless of the meaning), then we would match any `16.x` version of Ubuntu.
If you see any concerns regarding this simple heuristic - would be good to hear since I did not invest that much time understanding various versioning schemes.

Reply:
Yea, thinking about it a little bit more, I don't see any really bad side effects by just string matching the prefix. 

I think where I probably got hung up, was it wasn't clear to me OSRelease.VersionID wasn't referring to a specific version, but is more of a string match pattern. So maybe consider renaming OSRelease.VersionID to something like VersionPrefix would make it more clear that this doesn't represent a specific version, but is more of a string match.

Reply:
Yes, you're right. I see this coming from the fact that the requirements are represented by the same structure as the release information retrieved from the release/version/lsb_release where the semantics of the attributes was used from.